kubernetes/elastic-helm/configmap-elastic.yaml

```diff
@@ -21,6 +21,21 @@ data:
     extensions:
     processors:
       batch: {}
+      resource:
+        attributes:
+          - key: deployment.environment
+            value: "opentelemetry-demo"
+            action: upsert
+      attributes:
+        actions:
+          - key: http.url
+            action: extract
+            pattern: '^(?P<short_url>https?://[^/]+(?:/[^/]+)*)(?:/(?P<url_truncated_path>[^/?]+/[^/?]+))(?:\?|/?$)'
```

Comment:
What do you think of using the [URL OTTL function](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/pkg/ottl/ottlfuncs#url) in the transform processor to ensure all cases are covered? Note that this change would also provide semantic convention URL attributes (url.domand and url.path)

```yaml
  transform:
    error_mode: ignore
    trace_statements:
      - context: span
        statements:
          - merge_maps(attributes, URL(attributes["http.url"]), "upsert")
```

Reply:
I'll push this upstream but agree this would be useful.